javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/model/typesystem/LazyType.java

```diff
@@ -165,7 +194,13 @@ public boolean isAssignableBy(ResolvedType other) {
 
     @Override
     public boolean equals(Object o) {
-        return getType().equals(o);
+        if (o instanceof LazyType) {
+            // If is a LazyType then we should find the concrete value of the other before comparing
+            ResolvedType otherType = getConcreteType((LazyType) o);
```

Comment:
Assuming o is a LazyType (already casted) we should be able to write 

    ResolvedType otherType = o.getConcreteType();